Foxy libboost_serialization.so.1.71.0: error adding symbols: DSO missing

### Description

Foxy build fails due to libboost_serialization.so.1.71.0: error adding symbols: DSO missing from command line

### Your environment
* ROS Distro: Foxy
* OS Version: Ubuntu 20.04.3
* Source build
* foxy branch

### Steps to reproduce
Following the build instruction from https://moveit.ros.org/install-moveit2/source/

### Expected behaviour
Build should complete as expected.

### Actual behaviour
Build stops with the following console log:

### Backtrace or Console output

```
[Processing: moveit_planners_ompl, moveit_ros_benchmarks, moveit_ros_move_group]
--- stderr: moveit_planners_ompl
/usr/bin/ld: CMakeFiles/test_planning_context_manager.dir/test/test_planning_context_manager.cpp.o: undefined reference to symbol '_ZN5boost7archive6detail22archive_serializer_mapINS0_15binary_oarchiveEE6insertEPKNS1_16basic_serializerE'
/usr/bin/ld: /lib/x86_64-linux-gnu/libboost_serialization.so.1.71.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [ompl_interface/CMakeFiles/test_planning_context_manager.dir/build.make:326: ompl_interface/test_planning_context_manager] Error 1
make[1]: *** [CMakeFiles/Makefile2:386: ompl_interface/CMakeFiles/test_planning_context_manager.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
---
Failed   <<< moveit_planners_ompl [4min 21s, exited with code 2]
```

--- a/moveit_planners/ompl/ompl_interface/CMakeLists.txt
+++ b/moveit_planners/ompl/ompl_interface/CMakeLists.txt
@@ -42,6 +42,7 @@ ament_target_dependencies(${MOVEIT_LIB_NAME}
   OMPL
   Boost
 )
+target_link_libraries(${MOVEIT_LIB_NAME} Boost::serialization)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES COMPILE_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES LINK_FLAGS "${OpenMP_CXX_FLAGS}")
 

